api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -144,3 +144,13 @@ func (m *AzureManagedControlPlane) setDefaultAutoScalerProfile() {
 		m.Spec.AutoScalerProfile.SkipNodesWithSystemPods = (*SkipNodesWithSystemPods)(ptr.To(string(SkipNodesWithSystemPodsTrue)))
 	}
 }
+
+func (m *AzureManagedControlPlane) setDefaultOIDCIssuerProfile() {
+	if m.Spec.OIDCIssuerProfile == nil {
+		m.Spec.OIDCIssuerProfile = &OIDCIssuerProfile{}
```

Comment:
I think this thread covers this: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3973#pullrequestreview-1618500917

But, just to clarify, the reason we're doing it this way instead of the way that autoscaler does it above (if user doesn't declare a profile config we omit it from the spec entirely) is because this more explicit "is disabled" implementation removes edge cases from upgrade scenarios?

Reply:
@jackfrancis this comment summarizes it https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3973#discussion_r1324774308

Reply:
I think because autoscaler is a little different from other fields in that we want to send `nil` over the wire to AKS when not defined in the CAPZ spec. OIDC issuer is more like other fields where we always send a non-nil value.